src/main/groovy/com/readytalk/gradle/plugins/ci/BuildNumberVersionPlugin.groovy

```diff
@@ -0,0 +1,26 @@
+package com.readytalk.gradle.plugins.ci
+
+import com.readytalk.gradle.util.PluginUtils
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class BuildNumberVersionPlugin implements Plugin<Project>, PluginUtils {
+  Project project
+  private CiInfoExtension infoExt
+
+  void apply(final Project project) {
+    this.project = project
+    project.with {
+      plugins.apply(CiInfoPlugin)
+      this.infoExt = project.plugins.getPlugin(CiInfoPlugin).extension
```

Comment:
What about something like this:

    def infoPlugin = plugins.apply(CiInfoPlugin)
    this.infoExt = infoPlugin.extension